core/ipc/sink/kafka/server/src/main/java/org/opennms/core/ipc/sink/kafka/server/offset/KafkaOffsetProvider.java

```diff
@@ -221,8 +222,12 @@ public void shutdown() {
 
     }
 
+    public KafkaOffsetProvider() {
+        this(new OnmsKafkaConfigProvider(KafkaSinkConstants.KAFKA_CONFIG_SYS_PROP_PREFIX));
+    }
+
     public KafkaOffsetProvider(KafkaConfigProvider configProvider) {
-        this.configProvider = Objects.requireNonNull(configProvider);
+        this.configProvider = configProvider;
```

Comment:
The `configProvider` is still required, isn't it? If so, we should still have `Objects.requireNonNull(...)` as a NotNull check.